src/Data/Vec/Base.agda

```diff
@@ -167,6 +167,17 @@ fs ⊛* xs = fs >>= λ f → map f xs
 allPairs : ∀ {m n} → Vec A m → Vec B n → Vec (A × B) (m * n)
 allPairs xs ys = map _,_ xs ⊛* ys
 
+-- Diagonal
+
+diagonal : ∀ {n} → Vec (Vec A n) n → Vec A n
+diagonal [] = []
+diagonal (xs ∷ xss) = head xs ∷ diagonal (map tail xss)
+
+infixl 1 _>>=′_
+
+_>>=′_ : ∀ {n} → Vec A n → (A → Vec B n) → Vec B n
```

Comment:
Any reason not to call this `_>>=_` like we do in e.g. `Data.Maybe.Base`?

Reply:
There is already function called `_>>=_` in `Data.Vec.Base` defined as cartesian product. i.e. similar to `_>>=_` for `List`.
This is actually not a monad for `Vec`. However, agda's do-notation works syntactically. So this naming might be pretty useful.
Since diagonal-based one is genuine definition of monad for `Vec`, I also think the old `_>>=_` definition should yield it's name to new one, but it will cause breaking change.

Reply:
Hmm, so if we have two different possible definitions for the bind operator, and both require to be named `_>>=_` to work with do-notation we appear to be in trouble. Do notation doesn't work with `_>>=′_` right? So maybe, both `_>>=_` and this should be called something else more meaningful and the user can decide?

 I'm really not sure, and would appreciate other people's thoughts...

> but that's a breaking change.
We're currently working on v2.0 which can have breaking changes, but we want to be sure we're making the _right_ breaking changes!

Reply:
Maybe we can just define a sub module `Zip` and `Cartesian` so that each module has their own definition of `_>>=_`.

Reply:
That seems like an excellent suggestion :+1: